frontend/src/core/cells/cells.ts

```diff
@@ -214,17 +218,39 @@ const {
         : state.cellIds.topLevelIds.indexOf(cellId);
     const insertionIndex = before ? index : index + 1;
 
+    let newCellCode = code;
+
+    if (includeSelectionAsInitialCode) {
+      const id = cellId as CellId;
+      const editorView = state.cellHandles[id].current?.editorView;
+      if (editorView) {
+        const [highlighted, leftover] = extractHighlightedCode(editorView);
+        if (highlighted.length > 0) {
```

Comment:
i think the issue is that for markdown/sql, highlighted > 0. if you write a test for `extractHighlightedCode`, can we get verify that no selection -> no highlighted